torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2642,6 +2642,73 @@ def generator():
     return list(generator())
 
 
+def sample_inputs_nn_pad(op_info, device, dtype, requires_grad, mode, **kwargs):
+    assert mode in ('constant', 'reflect', 'replicate', 'circular')
+    if dtype == torch.bool and mode == 'circular':
+        # test_dtypes fails on ASAN with for the case below
+        # runtime error: load of value 190, which is not a valid value for type 'bool'
+        # Reference: https://github.com/pytorch/pytorch/pull/62814#issuecomment-894156562
+        shapes = ((1, 3), (2, 3, 3), (1, 3, 3), (3, 3, 3, 3), (3, 3, 5, 5))
+        pads = ((1, 2),)
+        negative_pad_case = ()
+    else:
+        # Supports 2-D, 3-D, 4-D, 5-D tensors
+        shapes = ((1, 3), (0, 3, 3), (1, 3, 3), (0, 3, 3, 3), (3, 3, 5, 5), (1, 3, 3, 3, 3))  # type: ignore[assignment]
+        pads = ((1, 2), (0, 1), (0, 2, 0, 1), (1, 1, 1, 1, 1, 1))  # type: ignore[assignment]
+        negative_pad_case = (
+            # (shape, pad)
+            ((1, 3, 4, 4), (-1, 1, -2, 1)),  # type: ignore[assignment]
+        )
+
+    make_inp = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    def generator():
+        if mode == 'constant':
+            # Default args
+            yield SampleInput(make_inp((1, 3, 3)), args=((2, 2),))
```

Comment:
So for constant pad we only have a single test case? Maybe constant pad should be a separate `sample_inputs_nn_pad_constant` function

Reply:
Actually it's more than that. 

Ref: https://github.com/pytorch/pytorch/pull/62814#discussion_r686171969